index.js

```diff
@@ -75,14 +82,25 @@ function toRegexRange(min, max, options) {
   return tok.result;
 }
 
-toRegexRange.cache = {};
-
 function siftPatterns(neg, pos, options) {
-  let onlyNegative = filterPatterns(neg, pos, '-', false, options) || [];
-  let onlyPositive = filterPatterns(pos, neg, '', false, options) || [];
-  let intersected = filterPatterns(neg, pos, '-?', true, options) || [];
+  let onlyNegative;
+  let onlyPositive;
+  let intersected;
+  if (options.antlr){
+    onlyNegative = filterPatterns(neg, pos, "'-'", false, options) || [];
```

Comment:
Seems like this would be better to do in `filterPatterns`, so it only needs to be done one time.